From 94bb101e2fa8e99ef3591860b096d2704cdbfa96 Mon Sep 17 00:00:00 2001
From: Eric Lundby <Eric.Lundby@gmail.com>
Date: Fri, 10 Apr 2026 10:52:02 -0600
Subject: [PATCH] use-conda-CMAKE_OSX_SYSROOT

/usr/bin/xcrun -sdk macosx --show-sdk-version returns the wrong SDK version on CI.

Use CMAKE_OSX_SYSROOT to get the correct SDK version.
---
 CMakeLists.txt |   17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -192,11 +192,18 @@ if(MLX_BUILD_METAL)
       OUTPUT_STRIP_TRAILING_WHITESPACE COMMAND_ERROR_IS_FATAL ANY)
     message(STATUS "Building on macOS ${MACOS_VERSION}")
 
-    # Get macOS SDK version.
-    execute_process(
-      COMMAND xcrun -sdk macosx --show-sdk-version
-      OUTPUT_VARIABLE MACOS_SDK_VERSION
-      OUTPUT_STRIP_TRAILING_WHITESPACE COMMAND_ERROR_IS_FATAL ANY)
+    # Get macOS SDK version. Prefer the SDK conda selects via CMAKE_OSX_SYSROOT;
+    # `xcrun --show-sdk-version` reports the wrong SDK on CI.
+    if(CMAKE_OSX_SYSROOT)
+      get_filename_component(_mlx_sdk_name "${CMAKE_OSX_SYSROOT}" NAME)
+      string(REGEX REPLACE "^MacOSX([0-9]+\\.[0-9]+)\\.sdk$" "\\1"
+                           MACOS_SDK_VERSION "${_mlx_sdk_name}")
+    else()
+      execute_process(
+        COMMAND xcrun -sdk macosx --show-sdk-version
+        OUTPUT_VARIABLE MACOS_SDK_VERSION
+        OUTPUT_STRIP_TRAILING_WHITESPACE COMMAND_ERROR_IS_FATAL ANY)
+    endif()
     if(MACOS_SDK_VERSION VERSION_LESS 14.0)
       message(
         FATAL_ERROR
